tests/test_transforms.py

```diff
@@ -172,18 +177,236 @@ def test_transformed_logprob(at_dist, dist_params, sp_dist, size):
+    a_value_var = a.clone()
+    a_value_var.name = "a_value"
+
+    b = at.random.normal(a, 1)
+    b.name = "b"
+    b_value_var = b.clone()
+    b_value_var.name = "b_value"
+
+    transform_opt = in2out(default_backward_transform, ignore_newtrees=True)
+    res = joint_logprob(
+        b, {a: a_value_var, b: b_value_var}, extra_rewrites=transform_opt
+    )
+    logp_vals_fn = aesara.function([a_value_var, b_value_var], res)
+
+    for a_val, b_val in itertools.product((-3, 0, 2), (-2, 0, 0.5)):
+        # logodds forward transform
+        transf_a_val = np.log(a_val / (1 - a_val))
+
+        logp_term_a = sp.stats.norm(0).logpdf(a_val)
+        # log jacobian of logodds  = log(expit(a) x (1 - expit(a))
+        expit_a = sp.special.expit(transf_a_val)
+        log_jacob_term_a = np.log(expit_a) + np.log1p(-expit_a)
+
+        logp_term_b = sp.stats.norm(a_val).logpdf(b_val)
+
+        total_logp = logp_term_a + log_jacob_term_a + logp_term_b
+        assert np.isclose(logp_vals_fn(transf_a_val, b_val), total_logp)
+
+
+@pytest.mark.parametrize("size", [None, 1, 2, (3, 2)])
+def test_scalar_loc_transform(size):
+    loc = at.scalar("loc")
+    y_rv = loc + at.random.normal(0, 1, name="y_rv", size=size)
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y}).sum()
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([loc, y], logp)
+
+    loc_test_val = 4.0
+    y_test_val = np.full(size, 1.0)
+
+    assert np.isclose(
+        logp_fn(loc_test_val, y_test_val),
+        sp.stats.norm(loc_test_val, 1).logpdf(y_test_val).sum(),
+    )
+
+
+@pytest.mark.parametrize("size", [None, 1, 2, (3, 2)])
+def test_scalar_scale_transform(size):
+    scale = at.scalar("scale")
+    y_rv = at.random.normal(0, 1, name="y_rv", size=size) * scale
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y}).sum()
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([scale, y], logp)
+
+    loc_test_val = 4.0
+    y_test_val = np.full(size, 1.0)
+
+    assert np.isclose(
+        logp_fn(loc_test_val, y_test_val),
+        sp.stats.norm(0, loc_test_val).logpdf(y_test_val).sum(),
+    )
+
+
+@pytest.mark.parametrize("size", [None, 1, 2, (3, 2)])
+def test_scalar_affine_transform(size):
+    loc = at.scalar("loc")
+    scale = at.scalar("scale")
+    y_rv = loc + at.random.normal(0, 1, name="y_rv", size=size) * scale
+    y = y_rv.type()
+    y.name = "y"
+
+    logp = joint_logprob(y_rv, {y_rv: y}).sum()
+    assert_no_rvs(logp)
+    logp_fn = aesara.function([loc, scale, y], logp)
+
+    loc_test_val = 4.0
+    scale_test_val = 0.5
+    y_test_val = np.full(size, 1.0)
+
+    assert np.isclose(
+        logp_fn(loc_test_val, scale_test_val, y_test_val),
+        sp.stats.norm(loc_test_val, scale_test_val).logpdf(y_test_val).sum(),
+    )
+
+
+def test_double_scalar_loc_transform():
+    loc1 = at.scalar("loc1")
+    loc2 = at.scalar("loc2")
+    y_rv = loc1 + loc2 + at.random.normal(0, 1, name="y_rv")
```

Comment:
For some weird reason the rewrite fails when the two scalar `loc`s are placed before the `rv`. It complains that I am trying to replace the original scalar `rv` with a vector `value_var`...

It passes with any of the following:
```python
y_rv = loc1 + at.random.normal(0, 1, name="y_rv") + loc2
y_rv = at.random.normal(0, 1, name="y_rv") + loc1 + loc2
```